types/index.d.ts

```diff
@@ -342,20 +342,69 @@ export interface CookiesOptions {
 }
 
 /** [Documentation](https://next-auth.js.org/configuration/events) */
-export type EventType =
-  | "signIn"
-  | "signOut"
-  | "createUser"
-  | "updateUser"
-  | "linkAccount"
-  | "session"
-  | "error"
+export type EventCallback<MessageType = unknown> = (
+  message: MessageType
+) => Promise<void>
```

Comment:
```suggestion
export type EventCallback<MessageType> = (
  message: MessageType
) => Promise<void>
```

I think here it is OK to enforce a generic, as otherwise this type is quite useless.

Reply:
Yeah this type is purely to avoid having to copy-paste the extra characters of `=> Promise<void>`